aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandlerTest.java

```diff
@@ -58,23 +69,221 @@
+                            .channels(Channel.builder()
+                                    .id("channelId")
+                                    .uri("channelUri")
+                                    .eventPublishers(EventPublisher.ANOMALY_DETECTION)
+                                    .build())
+                            .build())
+                    .build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+            final ProgressEvent<ResourceModel, CallbackContext> response
+                    = subject.handleRequest(proxy, request, null, logger);
+
+            assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
+            assertThat(response.getCallbackContext()).isNull();
+            assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
+            assertThat(response.getResourceModel()).isEqualTo(request.getDesiredResourceState());
+            assertThat(response.getMessage()).isNull();
+            assertThat(response.getErrorCode()).isNull();
+        }
+
+        @Test
+        public void itOnlyCallsAddNotificationChannelWhenUpdatedModelHasAdditionalChannelsNoIdSet() {
+            doReturn(GetNotificationConfigurationResponse.builder()
+                    .notificationConfiguration(NotificationConfiguration.builder()
+                            .channels(Channel.builder()
+                                    .id("channelId")
+                                    .uri("channelUri")
+                                    .eventPublishers(EventPublisher.ANOMALY_DETECTION)
+                                    .build())
+                            .build())
+                    .build())
+                    .when(proxy).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+
+            // the new model should have the additional channel
+            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName).anomalyDetectionNotificationConfiguration(
+                    ImmutableList.of(
+                            software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                    .channelUri("channelUri")
+                                    .build(),
+                            software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                    .channelUri("channelUri2")
+                                    .build()))
+                    .build());
+
+            subject.handleRequest(proxy, request, null, logger);
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getPolicyRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(getNotificationConfigurationRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(addNotificationChannelsRequest), any());
```

Comment:
I suggest naming `addNotificationChannelsRequest` to reflect its contents: Someting like `addChannel2Request`?

Reply:
I'm confused -- hadn't the `addChannel2Request` been added in the previous commits? It seems to have been undone again?